src/prefect/tasks/aws/lambda_function.py

```diff
@@ -0,0 +1,407 @@
+            either zip_file or (bucket and bucket_key) must be passed
+        - bucket (str): an S3 bucket in the same AWS region as your function
+        - bucket_key (str): the Amazon S3 key of the deployment package
+        - object_version (str, optional): for versioned S3 objects, the version of the
+            deployment package to use
+        - description (str, optional): description of Lambda function
+        - timeout (int, optional): Lambda function timeout in seconds, default is 3 seconds
+        - memorysize (int, optional): amount of memory that Lambda function has
+            access to in MB, must be a multiple of 64 MB, default is 128
+        - publish (bool, optional): set to True to publish the first version of the
+            function during creation, defaults to True
+        - subnet_ids (List[str], optional): list of subnet ids for vpc
+            configuration
+        - security_group_ids (List[str], optional): list of security
+            group ideas for vpc configuration
+        - dead_letter_config (dict, optional): a dead letter queue configuration that
+            specifies the queue or topic where Lambda sends asynchronous events
+            when they fail processing
+        - environment_variables (dict, optional): key-value pairs of environment
+            variables to pass to the Lambda function
+        - kms_key_arn (str, optional): the ARN of the AWS key management service used
+            to encrypt your function's environment variables, if not provided, AWS
+            Lambda uses a default service key
+        - tags (dict, optional): a list of tags to apply to the function, string
+            to string map
+        - tracing_config (str, optional): set to Active to samle and trace a
+            subset of incoming requests with Amazon X-Ray
+        - layers (List[str], optional): a list of function layers to add to
+            the function's execution environment, specify each layer by its ARN
+        - aws_credentials_secret (str, optional): the name of the Prefect Secret
+            which stores your AWS credentials; this Secret must be a JSON string
+            with two keys: `ACCESS_KEY` and `SECRET_ACCESS_KEY`
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
+            Task constructor
+    """
+
+    def __init__(
+        self,
+        function_name: str,
+        runtime: str,
+        role: str,
+        handler: str,
+        zip_file: str = None,
+        bucket: str = "",
+        bucket_key: str = "",
+        object_version: str = None,
+        description: str = "",
+        timeout: int = 3,
+        memorysize: int = 128,
+        publish: bool = True,
+        subnet_ids: list = None,
+        security_group_ids: list = None,
+        dead_letter_config: dict = None,
+        environment_variables: dict = None,
+        kms_key_arn: str = "",
+        tags: dict = None,
```

Comment:
`tags` will collide with an argument on the base `Task` class

Reply:
So will `timeout` actually; maybe these could be changed to `lambda_tags` and `lambda_timeout`?

Reply:
Good catches! Both corrected in the current batch of commits